### DISCUSSION TITLE
set size limit for binary cache

### DISCUSSION THREAD

**Discussion by Be-ing** [microsoft/vcpkg#19452] [Ideas]
**Is your feature request related to a problem? Please describe.**
vcpkg adds to the binary cache continually without evicting old packages.

**Proposed solution**
Implement some way to specify the maximum cache size with an environment variable. I am not sure what the best way to do this would be. Setting a single size would not be sufficient because multiple cache sources can be specified in VCPKG_BINARY_SOURCES.

**Describe alternatives you've considered**
Periodically delete old packages server side, but there is no server for the default caching which just stores to the filesystem.

**Additional context**
Exceeding a size limit with a cache could require rebuilding all packages to get rid of old ones.

**Comment by autoantwort** [comment]
I thought of placing a file in the root of the filesystem binary cache. Do you simply want to specify a maximum size or maybe something like "up to 90% disk usage" or "until 5GB free space is left"

**Reply by autoantwort** [reply]
I have created https://github.com/microsoft/vcpkg-tool/pull/945 for that
